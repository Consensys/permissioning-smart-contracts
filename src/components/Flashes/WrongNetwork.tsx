// Libs
import React from 'react';
import PropTypes from 'prop-types';
import { Box, Grid, Typography } from '@material-ui/core';
import WarningIcon from '@material-ui/icons/Warning';

const WrongNetwork: React.FC<{ networkId: number }> = ({ networkId }) => (
  <Box>
    <Grid container alignItems="center">
      <WarningIcon />
      <Typography variant="body1" color="error">
        Connect to the network where the contracts are deployed.
      </Typography>
    </Grid>
    <Grid container alignItems="center">
      <div className="wrongNetworkMessage">
        <Typography variant="body1">
          Change your network using MetaMask.
          {networkId === 1
            ? " You're currently on the Main Ethereum Network."
            : networkId === 2
            ? " You're currently on the Morden Classic Test Network."
            : networkId === 3
            ? " You're currently on the Ropsten Test Network."
            : networkId === 4
            ? " You're currently on the Rinkeby Test Network."
            : networkId === 5
            ? " You're currently on the Goerli Test Network."
            : ` You're currently on unknown network of id ${networkId}.`}
        </Typography>
      </div>
    </Grid>
  </Box>
);

WrongNetwork.propTypes = {
  networkId: PropTypes.number.isRequired
};

export default WrongNetwork;
