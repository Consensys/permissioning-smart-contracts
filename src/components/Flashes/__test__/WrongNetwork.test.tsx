// Libs
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount, ReactWrapper, ShallowWrapper } from 'enzyme';
// Components
import WrongNetwork from '../WrongNetwork';

var errorMessages = [
  () => "Change your network using MetaMask. You're currently on the Main Ethereum Network.",
  () => "Change your network using MetaMask. You're currently on the Morden Classic Test Network.",
  () => "Change your network using MetaMask. You're currently on the Ropsten Test Network.",
  () => "Change your network using MetaMask. You're currently on the Rinkeby Test Network.",
  () => "Change your network using MetaMask. You're currently on the Goerli Test Network.",
  (networkId: number) => `Change your network using MetaMask. You're currently on unknown network of id ${networkId}.`
];

describe('<WrongNetwork />', () => {
  let wrapper: ReactWrapper<any, any, any>;

  beforeEach(() => {
    wrapper = mount(<WrongNetwork networkId={0} />);
  });

  it('contains an element of class wrongNetworkMessage', () => {
    expect(wrapper.find('div.wrongNetworkMessage')).toHaveLength(1);
  });

  it('has props networkId=0', () => {
    expect(wrapper.props().networkId).toBe(0);
  });

  describe('error message', () => {
    let increment = 1;
    let errorMessage: string;
    let wrongNetworkWrapper: ShallowWrapper;
    beforeEach(() => {
      wrongNetworkWrapper = shallow(<WrongNetwork networkId={increment} />);
      errorMessage = wrongNetworkWrapper.find('.wrongNetworkMessage').text();
    });

    errorMessages.forEach(expectedErrorMessage => {
      it('contains the appropriate text', () => {
        expect(errorMessage).toEqual(expectedErrorMessage(increment));
        increment++;
      });
    });

    it('matches snapshot', () => {
      expect(toJson(wrongNetworkWrapper)).toMatchSnapshot();
    });
  });
});
