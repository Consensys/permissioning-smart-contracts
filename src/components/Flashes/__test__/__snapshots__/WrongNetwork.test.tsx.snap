// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrongNetwork /> error message matches snapshot 1`] = `
<Styled(MuiBox)>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
  >
    <Memo(ForwardRef(WarningIcon)) />
    <WithStyles(ForwardRef(Typography))
      color="error"
      variant="body1"
    >
      Connect to the network where the contracts are deployed.
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
  >
    <div
      className="wrongNetworkMessage"
    >
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        Change your network using MetaMask.
         You're currently on unknown network of id 7.
      </WithStyles(ForwardRef(Typography))>
    </div>
  </WithStyles(ForwardRef(Grid))>
</Styled(MuiBox)>
`;
